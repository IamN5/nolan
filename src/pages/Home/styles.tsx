import styled from "styled-components";
import { Colors } from "./../../styles";

interface ContainerProps {
  width: number;
}

export const Container = styled.div<ContainerProps>`
  padding: ${props => (props.width < 822 ? "20px" : "40px")};
  text-align: left;
`;

export const Title = styled.div`
  font-size: 25px;
  font-weight: bold;
  color: ${Colors.black};
`;
