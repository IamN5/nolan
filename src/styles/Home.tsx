import styled from "styled-components";
import { Colors } from "./shared";

export const Container = styled.div`
    padding: 40px;
    text-align: left;
`;

export const Title = styled.div`
    font-size: 25px;
    font-weight: bold;
    color: ${Colors.black};
`;
