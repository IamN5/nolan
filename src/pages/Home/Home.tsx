import React from "react";
import { Container, Title } from "./styles";

interface HomeProps {
    width: number;
}

function Home(props: HomeProps) {
    return (
        <>
            <Container width={props.width}>
                <Title>New Releases</Title>
            </Container>
        </>
    );
}

export default Home;
