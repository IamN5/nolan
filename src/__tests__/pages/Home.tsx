import React from "react";
import Home from "../../pages/Home/Home";
import renderer from "react-test-renderer";

it("renders Home component", () => {
    const tree = renderer.create(<Home width={300} />).toJSON();
    expect(tree).toMatchSnapshot();
});
