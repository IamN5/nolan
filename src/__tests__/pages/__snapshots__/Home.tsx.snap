// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component 1`] = `
<div
  className="sc-bdVaJa dvqYlX"
  width={300}
>
  <div
    className="sc-bwzfXH phNPS"
  >
    New Releases
  </div>
</div>
`;
